CMakeLists.txt

```diff
@@ -395,25 +453,28 @@ if(LIBUV_BUILD_TESTS)
            WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 endif()
 
-if(UNIX)
+if (UNIX)
   # Now for some gibbering horrors from beyond the stars...
-  foreach(x ${uv_libraries})
-    set(LIBS "${LIBS} -l${x}")
-  endforeach(x)
+  foreach(lib IN LISTS uv_libraries)
+    list(APPEND LIBS "-l${lib}")
+  endforeach()
+  string(REPLACE ";" " " LIBS "${LIBS}")
+  #Consider setting project version via project() call?
   file(STRINGS configure.ac configure_ac REGEX ^AC_INIT)
-  string(REGEX MATCH [0-9]+[.][0-9]+[.][0-9]+ PACKAGE_VERSION "${configure_ac}")
-  string(REGEX MATCH ^[0-9]+ UV_VERSION_MAJOR "${PACKAGE_VERSION}")
+  string(REGEX MATCH "([0-9]+)[.][0-9]+[.][0-9]+" PACKAGE_VERSION "${configure_ac}")
+  set(UV_VERSION_MAJOR "${CMAKE_MATCH_1}")
   # The version in the filename is mirroring the behaviour of autotools.
-  set_target_properties(uv PROPERTIES VERSION ${UV_VERSION_MAJOR}.0.0
-				      SOVERSION ${UV_VERSION_MAJOR})
+  set_target_properties(uv PROPERTIES
+    VERSION ${UV_VERSION_MAJOR}.0.0
+    SOVERSION ${UV_VERSION_MAJOR})
   set(includedir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_INCLUDEDIR})
   set(libdir ${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR})
   set(prefix ${CMAKE_INSTALL_PREFIX})
-  configure_file(libuv.pc.in ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc @ONLY)
+  configure_file(libuv.pc.in libuv.pc @ONLY)
 
   install(DIRECTORY include/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
   install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DOCDIR})
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/libuv.pc
+  install(FILES ${PROJECT_BINARY_DIR}/libuv.pc
```

Comment:
Is PROJECT_BINARY_DIR correct when configure_file() is relative to CMAKE_CURRENT_BINARY_DIR?

Reply:
In this case, yes. `${PROJECT_BINARY_DIR}` and `${CMAKE_CURRENT_BINARY_DIR}` are the same.

Reply:
Okay, but why change it then? CMAKE_CURRENT_BINARY_DIR looks a little more self-evidently correct to me.

(Consulting the cmake docs didn't really give a satisfying answer.)

Reply:
This might have been part of a `%s/CMAKE_CURRENT_BINARY/PROJECT_BINARY/g` command in vim. 

`CMAKE_CURRENT_BINARY_DIR` is set to whatever is the active CMakeLists.txt file.
`PROJECT_BINARY_DIR` is set to the current directory that called `project()` (This means that child directories can use `PROJECT_SOURCE_DIR` and friends and not affect parent scopes)

Typically you *rarely* need `CMAKE_CURRENT`, unless you take the more `autoconf` approach to CMakeLists.txt by putting one in each directory. If I was writing functions for general usage for a larger project (something on the scale of LLVM or GNU Radio, some monorepo project, or a generic CMake library), it would be something I would take into account, but for something as "small" in focus as libuv it's perfectly harmless.

Reply:
Okay thanks, answer accepted. =)